### PR TITLE
use circleci 2.0

### PR DESCRIPTION
**Pull request by michaelshobbs**
last build time was 20:53. down from the last master build time of 1:47:35

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    parallelism: 4
+    steps:
+      - checkout
+      - run: |
+          make ci-dependencies
+      - run: |
+          ./tests/ci/setup.sh
+      - run: |
+          sudo apt-get remove -qq -y git
+          sudo apt-get install -qq -y git=1:1.9.1-1ubuntu0.6 git-man=1:1.9.1-1ubuntu0.6
```
Comment (josegonzalez) [inline .circleci/config.yml]: Woof but I get it.

File: tests/ci/setup.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  sudo -E CI=true make -e sshcommand
+  # need to add the dokku user to the docker group
+  sudo usermod -G docker dokku
+  [[ "$1" == "buildstack" ]] && BUILD_STACK=true make -e stack
+  sudo -E CI=true make -e install
+  sudo -E make -e setup-deploy-tests
+  bash --version
+  docker version
+  lsb_release -a
+  # setup .dokkurc
```
Comment (josegonzalez) [inline tests/ci/setup.sh]: Wait does dokku now require this directory? I thought it was optional?

File: .circleci/config.yml (hunk 1 above)
Comment (michaelshobbs) [reply .circleci/config.yml]: Yup....

File: tests/ci/setup.sh (hunk 2 above)
Comment (michaelshobbs) [reply tests/ci/setup.sh]: This has always been there. We place some CI specific settings there in `.circleci/config.yml`

**Pull request merged**